pkg/operator/ceohelpers/bootstrap.go

```diff
@@ -141,7 +139,11 @@ func CheckSafeToScaleCluster(
 		return fmt.Errorf("CheckSafeToScaleCluster unrecognized scaling strategy %q", scalingStrategy)
 	}
 
-	nodeCount := len(operatorStatus.NodeStatuses)
+	node, err := masterNodeLister.List(masterNodeSelector)
+	if err != nil {
+		return fmt.Errorf("CheckSafeToScaleCluster couldn't get master nodes: %w", err)
+	}
+	nodeCount := len(node)
```

Comment:
what if there are nodes that are not in ready state? I think the node status is slightly more advanced

Reply:
- please correct me if I missed something or not

1. I'm afraid the quorum check has never cared about the nodes ready or not , it just check the length of master nodes:
https://github.com/openshift/cluster-etcd-operator/blob/3d5483e1871ba147a692736c71645175a85769c4/pkg/operator/ceohelpers/bootstrap.go#L144-L148

2. the node_controller will add or remove the nodes to the operator.nodestatus just by watching the master nodes:
https://github.com/openshift/cluster-etcd-operator/blob/3d5483e1871ba147a692736c71645175a85769c4/vendor/github.com/openshift/library-go/pkg/operator/staticpod/controller/node/node_controller.go#L47-L86

3. and the operator.nodestatus may have not updated  when we use it in the quorum check

Reply:
What's the difference then? Besides that we have one less indirection via the node status controller? 

Below the range you've linked, there's logic for NodeReady which we're going to lose:

https://github.com/openshift/cluster-etcd-operator/blob/3d5483e1871ba147a692736c71645175a85769c4/vendor/github.com/openshift/library-go/pkg/operator/staticpod/controller/node/node_controller.go#L89-L102

Reply:
having said that, any logic based on the node and status is more explicitly encoded in the `MemberHealth` call a couple of lines below, so I think we can (IMHO) remove all node based logic. 

I would only try to add a timeout to the memberhealth call context, otherwise we might get stuck as we've seen multiple times with GRPC.